test/integration_tests/conftest.py

```diff
@@ -85,14 +87,16 @@ def pytest_addoption(parser):
 
 
 @pytest.fixture(autouse=True)
-def temp_hub_dir(tmpdir, pytestconfig):
+def temp_hub_dir(tmp_path, pytestconfig):
     if not pytestconfig.getoption("use_tmp_hub_dir"):
         yield
     else:
         org_dir = torch.hub.get_dir()
-        torch.hub.set_dir(tmpdir)
+        subdir = os.path.join(tmp_path, "hub")
+        torch.hub.set_dir(subdir)
         yield
         torch.hub.set_dir(org_dir)
+        shutil.rmtree(subdir)
```

Comment:
```suggestion
        shutil.rmtree(subdir, ignore_errors=True)
```

Just in case the `subdir` directory is never used/created.